### PR TITLE
Logs: improve deduplication supporting code and fix duplicated frames

### PR DESCRIPTION
**Pull request by matyax**
This PR does two things:

- Gets rid of the deduplication implementation based on custom refId with infinite scrolling prefix. We inverted the parameters of `decorateData` to pass the log decoration function that we need, with or without deduplication.
- Resolves an issue related with data frame combination that was resulting in, instead of combined frames, duplicated frames with the same refId. It was not noticeable by the user, only when using the table which renders frames.

Broken:

https://github.com/grafana/grafana/assets/1069378/778ae521-dc8b-47f5-9525-0505173c3ef8

Fixed:

https://github.com/grafana/grafana/assets/1069378/55db28e0-a30d-40dc-8b9a-7a35a4863411

### CODE REVIEW

**Comment by matyax** [comment]
@grafana/observability-logs I feel tempted to backport this one. What do you think?

File: packages/grafana-o11y-ds-frontend/src/combineResponses.ts [hunk 1]
```diff
@@ -159,9 +159,12 @@ function shouldCombine(frame1: DataFrame, frame2: DataFrame): boolean {
   // because we do not have a good "frametype" value for them yet.
   const customType1 = frame1.meta?.custom?.frameType;
   const customType2 = frame2.meta?.custom?.frameType;
-
+  // Legacy frames have this custom type
   if (customType1 === 'LabeledTimeValues' && customType2 === 'LabeledTimeValues') {
```
Comment (matyax) [inline packages/grafana-o11y-ds-frontend/src/combineResponses.ts:163]: See https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/frame.go#L141 and https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/frame.go#L209

Also this is very Loki specific and infinite scrolling is not Loki exclusive IIRC.

**Review by gtk-grafana — APPROVED** [review]
Looks good and works well locally. Fixes the bug when switching between logs and table, and you've already infinite scrolled. Dedupe appears to still work as expected when scrolling triggers more results as well.

**Comment by matyax** [comment]
The changes only affect infinite scrolling code that lives in Explore, so I'll merge, but please feel free to comment or open issues if you find any.

**Pull request merged**